kdb/keyset.go

```diff
@@ -74,19 +84,17 @@ func toCKeySet(keySet KeySet) (*ckeySet, error) {
 	ckeySet, ok := keySet.(*ckeySet)
 
 	if !ok {
-		// TODO REVIEW: What is a ckeySet? (Error message not helpful)
-		return nil, errors.New("only pointer to ckeySet struct allowed")
+		return nil, errors.New("only instances of KeySet that were created by elektra/kdb may be passed to this function")
 	}
 
 	return ckeySet, nil
 }
 
-// TODO REVIEW: Confusing description of Append
-
-// Append adds a KeySet and returns the new length of the KeySet
-// after appending or -1 if keySet is not a pointer of type ckeySet.
-func (ks *ckeySet) Append(keySet KeySet) int {
-	cKeySet, err := toCKeySet(keySet)
+// Append appends all Keys from `other` to this KeySet and returns the
+// new length of this KeySet or -1 if `other` is not a KeySet which was
+// created by elektra/kdb.
```

Comment:
How to create it not with Elektra?

Reply:
Every struct that adheres to the `Key` interface could theoretically be passed to this function.